decred/decred/dcr/blockchain.py

```diff
@@ -0,0 +1,276 @@
+            idx (int): The child index at which to start the scan.
+            gap (int): The maximum number of unseen addresses allowed before
+                quitting the scan.
+
+        Returns:
+            list(int): The discovered address indices.
+        """
+        discovered = []
+        netParams = self.netParams
+        end = idx + gap
+        lastSeen = idx
+
+        its = 0
+        startTime = time.time()
+        while end > idx:
+            its += 1
+            addrs = [
+                addrlib.deriveChildAddress(xPub, i, netParams)
+                for i in range(idx, end + 1)
+            ]
+            founds = self.rpc.existsAddresses(addrs)
+            for j, found in enumerate(founds):
+                if found:
+                    discovered.append(idx + j)
+                    lastSeen = idx + j
+            idx = end + 1
+            end = lastSeen + 1 + gap
+
+        duration = time.time() - startTime
+        log.debug(
+            f"{its} iterations to discover {len(discovered)} addresses. {duration:.3f} seconds"
+        )
+
+        return discovered
+
+    def syncAccount(self, addrs, outPoints, startHash):
+        """
+        Sync an account.
+
+        Args:
+            addrs list(str): A list of addresses for the transaction filter.
+            outPoints list(msgtx.OutPoint): A list of outpoints for the
+                transaction filter.
+            startHash (ByteArray): Scans will be started at this block. If
+                startHash is not specified, the scan will be started at the root
+                block, which is probably not the genesis block. See syncHeaders
+                documentation for more info on the root block.
+
+        Return:
+            list(RescanBlock): The discovered transactions grouped by block
+                hash.
+        """
+        self.rpc.loadTxFilter(True, addrs, outPoints)
+        maxBlocksPerRescan = 2000
+        mainchain = self.mainchainDB
+
+        startHeader = self.headerDB[startHash]
+        start = startHeader.height
+        tipHeight = mainchain.last()[0]
+        toScan = tipHeight - start + 1
+        log.info(f"synchronizing {toScan} blocks")
+        rescanBlocks = []
+        while start <= tipHeight:
+            blockHashes = [x[1] for x in mainchain[start : start + maxBlocksPerRescan]]
+            rescanBlocks.extend(self.rpc.rescan(blockHashes))
+            start += maxBlocksPerRescan
+
+        return rescanBlocks
+
+    def syncHeaders(self, root):
+        """
+        Synchronize headers after specified blockHash. It is assumed that root
+        block is a mainchain block, a condition easily met if wallets choose
+        their initial root block to be a few blocks before the tip at wallet
+        creation.
+
+        Args:
+            root (ByteArray): The block hash of the root.
+
+        Returns:
+            list(ByteArray): A list of hashes for blocks removed from mainchain.
+        """
+        synced = 0
```

Comment:
Could you change this to syncedHeight/syncedHt/syncedTo. It looks like an atomic boolean value at fist glance.